src/main/webapp/assets/js/view/application-add-wizard.js

```diff
@@ -680,10 +680,12 @@ define([
                 this.$('.required-config-loading').hide()
                 if (catalogEntryItem!=null && catalogEntryItem.config!=null) {
                     var that = this
-                    _.each(catalogEntryItem.config, function (cfg) {
-                        if (cfg.priority !== undefined) {
+                    _.chain(catalogEntryItem.config).sortBy(function (cfg){
+                        return !isNaN(cfg.priority) ? cfg.priority : 99999;
+                    }).each(function (cfg) {
+                        if (cfg.pinned === true || (cfg.contraints.indexOf('required') > -1 && typeof cfg.defaultValue === 'string' && cfg.defaultValue === '')) {
```

Comment:
Sorry but it's not possible. In case of `defaultValue == null` in java, [it is not serialised](https://github.com/apache/brooklyn-server/blob/master/rest/rest-api/src/main/java/org/apache/brooklyn/rest/domain/ConfigSummary.java#L45-L46) therefore, the only possibilities are either `undefined` or a non empty string.